pkg/controller/pod.go

```diff
@@ -131,6 +139,25 @@ func (pnc *PodNetworksController) worker() {
 	}
 }
 
+func (pnc *PodNetworksController) reconcileOnStartup() error {
+	pods, err := pnc.podsLister.List(labels.Everything())
+	if err != nil {
+		err = fmt.Errorf("failed to list pods on current node: %v", err)
+		return err
+	}
+	for _, pod := range pods {
+		// since there is no such "not has" relation in a field selector,
+		// filter out pods that are of no concern to the controller here
+		if pod.Spec.HostNetwork {
+			continue
```

Comment:
would we want to log these ?

Reply:
Yeah we could log an entry at debug level for troubleshooting, however a better approach I think would be to delete lines 149-153 and filter out pods with hostnetwork in `processNextWorkItem` function where we can logs a warning and throw an event as a feedback to the user in the case the latter added a network annotation to these pods as suggested in the comment no 2 https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/89#discussion_r1027361540

Will update this PR based on the decision we take on this subject.

Reply:
I understand your point, but, I wonder if it isn't more efficient to filter the events we send to the queue instead.

We could move all this code to a func, and call from both places (here and in the handlePodUpdate).

I'd say something like:
```golang
func ignoreHostNetworkedPods(pod) {
    if pod.Spec.HostNetwork {
        klog.Warningf(
			"rejecting to add interfaces for host networked pod: %s",
			annotations.NamespacedName(newPod.GetNamespace(), newPod.GetName()),
		)
	pnc.Eventf(
	    newPod,
	    corev1.EventTypeWarning, 
	    "InterfaceAddRejected", 
	    rejectInterfaceAddEventFormat(newPod))
        continue
    }
}
```
Does this make sense ? I have no numbers to backup my claim ..

Reply:
sure, let do this.

Will wait for https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/89 to be merged with the refactored code that uses the new `ignoreHostNetworkedPods` function and update this PR accordingly